stl/inc/xstring

```diff
@@ -3866,11 +3863,12 @@ public:
         const size_type _Old_size    = _Mypair._Myval2._Mysize;
         const size_type _Suffix_size = _Old_size - _Nx - _Off + 1;
         if (_Count < _Nx) { // suffix shifts backwards; we don't have to move anything out of the way
-            _Mypair._Myval2._Mysize = _Old_size - (_Nx - _Count);
             _Elem* const _Old_ptr   = _Mypair._Myval2._Myptr();
             _Elem* const _Insert_at = _Old_ptr + _Off;
             _Traits::move(_Insert_at, _Ptr, _Count);
             _Traits::move(_Insert_at + _Count, _Insert_at + _Nx, _Suffix_size);
+            _ASAN_STRING_MODIFY(*this, _Old_size, _Old_size - (_Nx - _Count));
+            _Mypair._Myval2._Mysize = _Old_size - (_Nx - _Count);

```

Comment:
Suggestion: This mentions `_Old_size - (_Nx - _Count)` twice. Could we extract to `_New_size`?